### PR TITLE
Add headers for more security

### PR DESCRIPTION
**Pull request by JoeChapman**
- X-Frame-Options so browser cannot render page in a frame
- X-Content-Tyoe-Options to disable content type sniffing
- X-XSS-Protection to ensure the cross-site scripting browser filter is enabled
- Strict-Transport-Security to instruct the browser to only access a web
application over a secure connection and and remember it for 12 months
- Cache-control to stop the browser of intermediary from caching sensitive data

### CODE REVIEW

**Comment by daniel-ac-martin** [comment]
This will be a breaking change we should version accordingly so that no-one accidentally breaks their service.

I wonder if the cache-control header is a bit overzealous. Do we want this to apply to static assets? Should this be set in the apps instead?

**Comment by JoeChapman** [comment]
@daniel-ac-martin I agree with your concern around the cache-control header. I wanted some strong opinions otherwise.

**Comment by JoeChapman** [comment]
image

File: nginx.conf [hunk 1]
```diff
@@ -79,20 +79,45 @@ http {
...
+
+    # config to enable HSTS(HTTP Strict Transport Security) https://developer.mozilla.org/en-US/docs/Security/HTTP_Strict_Transport_Security
+    # to avoid ssl stripping https://en.wikipedia.org/wiki/SSL_stripping#SSL_stripping
+    add_header Strict-Transport-Security "max-age=31536000; includeSubdomains;";
+
+
+    location ~* \.(?:manifest|appcache|html?|xml|json)$ {
+      # Disable caching for manifest, appcache, html, xml and json files
+      add_header Cache-control: no-store, no-cache
+    }
```
Comment (daniel-ac-martin) [inline nginx.conf]: I vote we remove this block. I think it makes the proxy is bit too clever for its own good. In general the apps should decide which responses are not cacheable. (Some HTML is cacheable.)

Most of our apps don't add the .html suffix to their pages so they would need to do this anyway.

Comment (JoeChapman) [inline nginx.conf]: That's a fair point. The block is a bit too specific to be so broadly applied

**Comment by allandegnan** [comment]
Seems fine.  You have tested this somewhere though?

**Comment by JoeChapman** [comment]
@allandegnan Tested with Docker-Compose

**Comment by JoeChapman** [comment]
Tested: https://drone.digital.homeoffice.gov.uk/UKHomeOffice/kube-end-tenancy/131

**Comment by allandegnan** [comment]
Bump the version number and push that tag

**Comment by JoeChapman** [comment]
@allandegnan Merge this first?

**Pull request merged**